### PR TITLE
Parcel printing response fix

### PR DESCRIPTION
**Pull request by ErnestasLuza**
Parcel reponse fix for deserialize to reponse class. Tested on ps 1.7 and ps8 working as intended also tested in Order list and in Order.

### CODE REVIEW

File: src/Api/Request/ParcelPrint.php [hunk 1]
```diff
@@ -52,7 +52,15 @@ public function printParcel(ParcelPrintRequest $request)
             );
         }
 
-        $responseBody = $serializer->deserialize($response, ParcelPrintResponse::class);
+        if (!is_array($response)) {
+            $decodedResponse = json_decode($response);
+            if (!isset($decodedResponse->status)) {
+                $decodedResponse['pdf'] = $response;
+                $responseBody = $serializer->deserialize($decodedResponse['pdf'], ParcelPrintResponse::class);
```
Comment (margud) [inline src/Api/Request/ParcelPrint.php]: i think you mean 
      $responseBody = $serializer->deserialize($decodedResponse, ParcelPrintResponse::class);

File: src/Api/Request/ParcelPrint.php [hunk 2]
```diff
@@ -52,7 +52,15 @@ public function printParcel(ParcelPrintRequest $request)
             );
         }
 
-        $responseBody = $serializer->deserialize($response, ParcelPrintResponse::class);
+        if (!is_array($response)) {
+            $decodedResponse = json_decode($response);
+            if (!isset($decodedResponse->status)) {
+                $decodedResponse['pdf'] = $response;
```
Comment (margud) [inline src/Api/Request/ParcelPrint.php]: so what kind of object it is not? after             $decodedResponse = json_decode($response); there is already something inside this object and then you add pdf.
wont in look like [something, 'pdf' => somethig]?

File: src/Api/Request/ParcelPrint.php [hunk 3]
```diff
@@ -52,7 +52,15 @@ public function printParcel(ParcelPrintRequest $request)
...
+                $responseBody = $serializer->deserialize($decodedResponse['pdf'], ParcelPrintResponse::class);
+            } else {
+                $responseBody = $serializer->deserialize($response, ParcelPrintResponse::class);
+            }
+        }
```
Comment (margud) [inline src/Api/Request/ParcelPrint.php]: i would just make something like this:
```
        if (!is_array($response)) {
            $decodedResponse = json_decode($response);
            if (!isset($decodedResponse->status)) {
                $responseBody = $serializer->deserialize(['pdf' => $response], ParcelPrintResponse::class);
            } else {
                $responseBody = $serializer->deserialize($response, ParcelPrintResponse::class);
            }
        }
```
        
        what do you think?

File: src/Api/Request/ParcelPrint.php [hunk 4]
```diff
@@ -52,7 +52,8 @@ public function printParcel(ParcelPrintRequest $request)
             );
         }
 
-        $responseBody = $serializer->deserialize($response, ParcelPrintResponse::class);
+        $responseBody = $serializer->deserialize(is_object($response) ? $response : json_decode($response),
```
Comment (margud) [inline src/Api/Request/ParcelPrint.php:55]: where is the 'pdf' => part xd now you just add string like it was before no?

**Review by margud — APPROVED** [review]

File: src/Api/Request/ParcelPrint.php (hunk 4 above)
Comment (ErnestasLuza) [reply src/Api/Request/ParcelPrint.php:55]: For future tip if somebody gonna review this PR all comments were answered in Slack.

**Pull request merged**